src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTargetScaler.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB.Trigger
+{
+    internal class CosmosDBTargetScaler : ITargetScaler
+    {
+        public const int DefaultMaxItemsPerInvocation = 100;
+        private readonly string _functionId;
+        private readonly TargetScalerDescriptor _targetScalerDescriptor;
+        private readonly CosmosDBMetricsProvider _cosmosDBMetricsProvider;
+        private readonly ILogger _logger;
+        private readonly CosmosDBTriggerAttribute _cosmosDBTriggerAttribute;
+        private readonly Container _monitoredContainer;
+
+        internal CosmosDBTargetScaler(string functionId, CosmosDBTriggerAttribute cosmosDBTriggerAttribute, Container monitoredContainer, Container leaseContainer, string processorName, ILogger logger)
+        {
+            _functionId = functionId;
+            _targetScalerDescriptor = new TargetScalerDescriptor(functionId);
+            _monitoredContainer = monitoredContainer;
+            _cosmosDBMetricsProvider = new CosmosDBMetricsProvider(logger, _monitoredContainer, leaseContainer, processorName);
+            _logger = logger;
+            _cosmosDBTriggerAttribute = cosmosDBTriggerAttribute;
+        }
+
+        public TargetScalerDescriptor TargetScalerDescriptor => _targetScalerDescriptor;
+
+        public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            CosmosDBTriggerMetrics metrics = await _cosmosDBMetricsProvider.GetMetricsAsync();
+
+            return GetScaleResultInternal(context, metrics.RemainingWork, metrics.PartitionCount);
+        }
+
+        internal TargetScalerResult GetScaleResultInternal(TargetScalerContext context, long remainingWork, int partitionCount)
+        {
+            int concurrency;
+
+            if (!context.InstanceConcurrency.HasValue)
+            {
+                if (_cosmosDBTriggerAttribute.MaxItemsPerInvocation > 0)
```

Comment:
concurrency = _cosmosDBTriggerAttribute.MaxItemsPerInvocation > 0 ? _cosmosDBTriggerAttribute.MaxItemsPerInvocation : DefaultMaxItemsPerInvocation;
```suggestion
                if (_cosmosDBTriggerAttribute.MaxItemsPerInvocation > 0)
```